src/plugins/discover/public/application/apps/main/components/doc_table/components/pager/tool_bar_pagination.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPagination,
+  EuiPopover,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n/';
+
+interface ToolBarPaginationProps {
+  pageSize: number;
+  pageCount: number;
+  activePage: number;
+  onPageClick: (page: number) => void;
+  onPageSizeChange: (size: number) => void;
+}
+
+export const ToolBarPagination = ({
+  pageSize,
+  pageCount,
+  activePage,
+  onPageSizeChange,
+  onPageClick,
+}: ToolBarPaginationProps) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const rowsWord = i18n.translate('discover.docTable.rows', {
+    defaultMessage: 'rows',
+  });
+
+  const onChooseRowsClick = () => setIsPopoverOpen((prevIsPopoverOpen) => !prevIsPopoverOpen);
```

Comment:
wouldn't just `setIsPopoverOpen(!isPopoverOpen)` work here?

Reply:
It should works, but React recommends using setState with callback in case the next state depends on the current one.